src/Progress/DeviceProgressBar.tsx

```diff
@@ -0,0 +1,93 @@
+import { ProgressBarContainer, TopText, CenterText, RightText, ProgressBar, Filler, BottomText } from './deviceProgressBarStyles';
+import { ThemeProvider } from '@emotion/react';
+import customTheme from 'src/styles/theme';
+
+export interface DeviceProgressProps {
+    /**
+     * This is to set the used space of the device. The type should be a number.
+     */
+    used: number;
+    /**
+     * This is to set the available space of the device. The type should be a number.
+     */
+    available: number;
+    /**
+     * This is to set the total space of the device. The type should be a number.
+     */
+    total: number;
+    /**
+     * This is to set the capacity for the device progress bar.
+     */
+    capacity: string;
+    /**
+     * This is to set the background color of the  progress bar.
+     */
+    backgroundColor?: string;
+}
+
+export const formatBytes = (bytes: any, decimals = 2) => {
```

Comment:
It appears `bytes` should be either of type `number | undefined` based on it's implementation.